code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -76,6 +76,10 @@
 	if(!cell)
 		return
 	cell.use(amount)
+	if(cell.rigged)
```

Comment:
I feel like this could be an issue for more things than just the batons. Defibs as well for example. Heck even for borgs. I feel like this needs a more generic solution (which might involve signals)